### PR TITLE
Reduce qutip import time by setting logger names

### PR DESCRIPTION
**Pull request by eendebakpt**
**Description**
Importing qutip takes a long time. On a Windows system import takes almost 4 seconds. The import time can be analyzed with [importtime-waterfall](https://github.com/asottile/importtime-waterfall). A large part of the time is spend in importing external packages (scipy, IPython, matplotilib), but also qutip itself adds to the import time.

This PR reduces the import time by setting the logger names in 4 submodules explicitly. 
Benchmark:
```
python -c "import time; import matplotlib; import IPython; import scipy; t0=time.time(); import qutip; dt=time.time()-t0; print(f'{dt*1e3:.2f} ms')"
``` 
Results:
```
dev.major: 1996.52 ms
PR: 1510.06 ms
```

### CODE REVIEW

File: doc/changes/1980.feature.py [hunk 1]
```diff
@@ -0,0 +1 @@
+Improve qutip import times
```
Comment (hodgestar) [inline doc/changes/1980.feature.py:1]:
```suggestion
Improve qutip import times by setting logger names explicitly.
```

**Review by hodgestar — CHANGES_REQUESTED** [review]
Thanks @eendebakpt. This is a great improvement for a small change. I left one suggestion for adding a bit more of a description to the change log, but I think it looks good to merge. Are there any more changes you'd like to make?

**Comment by hodgestar** [comment]
For QuTiP v5 we're thinking of implementing https://scientific-python.org/specs/spec-0001/ (i.e. for the dev.major branch) but we haven't gotten around to it yet.

**Comment by eendebakpt** [comment]
> Thanks @eendebakpt. This is a great improvement for a small change. I left one suggestion for adding a bit more of a description to the change log, but I think it looks good to merge. Are there any more changes you'd like to make?

I am creating a similar PR against dev.major. Perhaps there are more optimizations to be done (have not checked yet), but I think it is better to put it in a seperate PR.

**Comment by hodgestar** [comment]
@eendebakpt Happy for me to merge this PR?

**Review by hodgestar — APPROVED** [review]
Thanks @eendebakpt.

**Pull request merged**